desc/compute/utils.py

```diff
@@ -0,0 +1,306 @@
+import warnings
+from termcolor import colored
+
+from desc.backend import jnp
+from desc.grid import LinearGrid
+
+
+def _get_proper_surface(grid, surface_label):
+    """Returns grid quantities associated with the given surface label.
+
+    Parameters
+    ----------
+    grid : Grid
+        Collocation grid containing the nodes to evaluate at.
+    surface_label : str
+        The surface label of rho, theta, or zeta.
+
+    Returns
+    -------
+    :rtype: (ndarray, ndarray, ndarray)
+    surface_label_nodes : ndarray
+        The column in the grid corresponding to this surface_label's nodes.
+    unique_idx : ndarray
+        The indices of the unique values of the surface_label in grid.nodes.
+    ds : ndarray
+        The differential elements (dtheta * dzeta for rho surface).
+    """
+    assert surface_label in {"rho", "theta", "zeta"}
+    if surface_label == "rho":
+        surface_label_nodes = grid.nodes[:, 0]
+        unique_idx = grid.unique_rho_idx
+        ds = grid.spacing[:, 1:].prod(axis=1)
+    elif surface_label == "theta":
+        surface_label_nodes = grid.nodes[:, 1]
+        unique_idx = grid.unique_theta_idx
+        ds = grid.spacing[:, [0, 2]].prod(axis=1)
+    else:
+        surface_label_nodes = grid.nodes[:, 2]
+        unique_idx = grid.unique_zeta_idx
+        ds = grid.spacing[:, :2].prod(axis=1)
+
+    return surface_label_nodes, unique_idx, ds
+
+
+def compress(grid, x, surface_label="rho"):
+    """Compress x by returning only the elements at unique surface_label indices.
+
+    Parameters
+    ----------
+    grid : Grid
+        Collocation grid containing the nodes to evaluate at.
+    x : ndarray
+        The array to compress.
+        Should usually represent a surface function (a function constant over a surface)
+        in an array that matches the grid's pattern.
+    surface_label : str
+        The surface label of rho, theta, or zeta.
+
+    Returns
+    -------
+    ndarray
```

Comment:
should give this some sort of name, otherwise the autodoc treats ndarray as the label and nothing for dtype